openlcs/libs/sc_handler.py

```diff
@@ -194,3 +219,116 @@ def unpack_source_container_image(self):
+        if comp_type == 'GOLANG':
+            dep_dir = os.path.join(
+                extra_src_dir, "deps", 'gomod', "pkg",
+                "mod", "cache", "download")
+            vendor_dir = os.path.join(extra_src_dir, "app", "vendor")
+            # Exist new version, such as:
+            # 'deps/gomod/pkg/mod/cache/download/k8s.io/klog/@v/v1.0.0.zip'
+            # 'deps/gomod/pkg/mod/cache/download/k8s.io/klog/v2/@v/v2.9.0.zip'
+            search_patterns = [
+                dep_dir + name_pattern + "/@v" + version_pattern + '.zip',
+                dep_dir + name_pattern + "/**/@v" + version_pattern + '.zip',
+                vendor_dir + search_name
+                ]
+        elif comp_type in ['NPM', 'YARN', 'PYPI']:
+            # mapping Python source between Corgi and OSBS
+            comp_type = 'pip' if comp_type == 'PYPI' else comp_type.lower()
+            dep_dir = os.path.join(extra_src_dir, 'deps', comp_type)
+            common_pattern = dep_dir + name_pattern + version_pattern
+            search_patterns = [common_pattern + extension
+                               for extension in ['.tgz', '.tar.gz', '.zip']]
+        return search_patterns
+
+    def get_remote_source_path(self, component, extra_src_dir):
+        """
+        Get source tarball path of the remote source component.
+        """
+        search_patterns = self.get_remote_source_search_patterns(
+            component, extra_src_dir)
+
+        # Search remote source path.
+        comp_path = search_content_according_patterns(search_patterns)
+        return comp_path
+
+    def get_container_remote_source(self, components):
+        """
+        Get remote source in source container.
+        """
+        missing_components = []
+        extra_src_dir = os.path.join(self.dest_dir, 'extra_src_dir')
+
+        # Get source for each component.
+        for component in components:
+            comp_path = self.get_remote_source_path(component, extra_src_dir)
+            name_version = get_component_name_version_combination(component)
+            if not comp_path:
+                missing_components.append(component)
+                continue
+            # Handle source when found it in app vendor, compress the source
+            # as a source tarball.
+            elif os.path.isdir(comp_path):
+                try:
+                    dest_path = os.path.join(
+                        comp_path, name_version + ".tar.gz")
+                    compress_source_to_tarball(dest_path, comp_path)
+                    comp_path = dest_path
+                except RuntimeError as err:
+                    err_msg = (f"Failed to compress {component} source in "
+                               f"app vendor: {err}")
+                    raise RuntimeError(err_msg) from None
+
+            # Create a directory to store remote source tarball.
+            comp_dir = os.path.join(self.dest_dir, 'rs_dir', name_version)
```

Comment:
Yes, move the source files to a unified directory, so that we can only use the unified directory(`rs_dir`) in the post step, 
if we do not move source files, should add the detailed source tarball path for each component in the post step.

Reply:
I think it is the same as misc data, we also move them to `metadata` directory, and it is also lose the original file path

Reply:
After checking, the path is from the source tarball, not related to the original source tarball path